contracts/modules/V3ToV4Migrator.sol

```diff
@@ -20,4 +28,38 @@ abstract contract V3ToV4Migrator is MigratorImmutables {
         return caller == owner || V3_POSITION_MANAGER.getApproved(tokenId) == caller
             || V3_POSITION_MANAGER.isApprovedForAll(owner, caller);
     }
+
+    /// @dev check that the v4 position manager call is a safe call
+    /// of the position-altering Actions, we only allow Actions.MINT
+    /// this is because, if a user could be tricked into approving the UniversalRouter for
+    /// their position, an attacker could take their fees, or drain their entire position
+    function _checkV4PositionManagerCall(bytes calldata inputs) internal view {
+        bytes4 selector;
+        assembly {
+            selector := calldataload(inputs.offset)
+        }
+        if (selector != V4_POSITION_MANAGER.modifyLiquidities.selector) {
+            revert InvalidAction(selector);
+        }
+
+        // slice is `abi.encode(bytes unlockData, uint256 deadline)`
+        bytes calldata slice = inputs[4:];
+        // the first bytes(0) extracts the unlockData parameter from modifyLiquidities
+        // unlockData = `abi.encode(bytes actions, bytes[] params)`
+        // the second bytes(0) extracts the actions parameter from unlockData
+        bytes calldata actions = slice.toBytes(0).toBytes(0);
+
+        uint256 numActions = actions.length;
+
+        for (uint256 actionIndex = 0; actionIndex < numActions; actionIndex++) {
+            uint256 action = uint8(actions[actionIndex]);
+
+            if (
+                action == Actions.INCREASE_LIQUIDITY || action == Actions.DECREASE_LIQUIDITY
```

Comment:
IMO it would be clearer (and should save some bytecode size) to express this condition as:

`if (action != Actions.MINT_POSITION) {`

Reply:
Or is there an explicit desire to allow all the other actions, and just disallow these three? In that case I'd argue the error name is a bit misleading and be `ForbiddenAction()` or something like that.

Reply:
Thinking about it more, it's gotta be the second case, so ignore the original comment and consider an error naming tweak.

Reply:
Yeah because we still want settling, taking, clearing as actions. We can definitely make the error better and leave a comment to make this clearer

Reply:
alternative check is `if (action < Actions.SETTLE && action != Actions.MINT_POSITION)`, not sure if it'll be cheaper